astropy/io/fits/column.py

```diff
@@ -1423,12 +1423,13 @@ def _init_from_array(self, array):
 
             # Check for unsigned ints.
             bzero = None
-            if 'I' in format and ftype.base == np.dtype('uint16'):
-                bzero = np.uint16(2**15)
-            elif 'J' in format and ftype.base == np.dtype('uint32'):
-                bzero = np.uint32(2**31)
-            elif 'K' in format and ftype.base == np.dtype('uint64'):
-                bzero = np.uint64(2**63)
+            if ftype.base.kind == 'u':
+                if 'I' in format:
+                    bzero = np.uint16(2**15)
+                elif 'J' in format:
+                    bzero = np.uint32(2**31)
+                elif 'K' in format:
+                    bzero = np.uint64(2**63)
```

Comment:
Sorry for not replying sooner, I wanted to check the meaning of removing the itemsize comparison. The format letters are computer above (``from_recformat``) from the dtype, so it is actually a good things to remove this! Thanks @subagonsouth .